### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by aliabbani**
## React capstone project - Metrics webapp

In this milestone we:

- Used correct Gitflow.
- The app is a single-page application (SPA) built with React and Redux.
- We find [Covid](https://covid19tracking.narrativa.com/index_en.html) API to base the development of the app around it.
- When the home page loads, the webapp shows the list of countries with numeric values for the historical total cases.
- When the user clicks (or taps) on a country item, the application navigates to the details page.
- When the details page loads, the webapp shows detailed data for a specific country.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- Created tests for the application.
- Styled your components to match the design provided.

[Project video Link](https://drive.google.com/file/d/1AS8uwawW_O2ObDgGVv_ed8ZgBfiFme9p/view) 

We Also:
- Added a new repo
- Created react-redux app
- Added linters
- Added reducer, store, and components.
- Added CSS file
- Added test

### CODE REVIEW

File: src/stylesheet/index.css [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  grid-template-columns: 1fr 1fr;
+  background-color: #f8d6e4;
+}
+
+.inside-flex,
+.inside-flex-bolder {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  align-items: end;
```
Comment (henatan99) [inline src/stylesheet/index.css]: - [x] The `align-item` property `end` has mixed support so preventing the app from running. I think you can use the `flex-end` property. Kindly make the required change here. :pray: 

Screenshot from 2021-10-14 17-20-18

File: README.md [hunk 2]
```diff
@@ -29,42 +47,37 @@ Your app is ready to be deployed!
...
+- Inspiration
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
+## 📝 License
 
-### `npm run build` fails to minify
+This project is [Microverse](https://www.microverse.org/) licensed.
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+Original design idea by Nelson Sakwa on [Behance](https://www.behance.net/gallery/31579789/Ballhead-App-%28Free-PSDs%29).
```
Comment (henatan99) [inline README.md]: - [x] I think you should take this line under the acknowledgment section. :pray:

File: src/components/Home.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+              <NavLink className="nav-icon" to={{ pathname: test, data }} exact>
+                <AiOutlineRightCircle className="inside-up" />
+              </NavLink>
+              <div className="inside-down">
+                <h3 className="inside-h3">{data.name}</h3>
+                <div className="confirmed">
+                  <h4 className="inside-h4">{data.today_confirmed.toLocaleString()}</h4>
+                </div>
+              </div>
+            </div>
```
Comment (henatan99) [inline src/components/Home.js]: - [x] I think this card component can be a separate component that can be imported and rendered here. React is a component-based framework and smaller components are more convenient for reusability, fixing bugs and errors and app scalability. :pray:

File: src/stylesheet/index.css [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+  color: #f7f7f8de;
+}
+
+.inside-flex {
+  background-color: #de4783;
+}
+
+.inside-flex-bolder {
+  background-color: #d1447a;
+}
```
Comment (henatan99) [inline src/stylesheet/index.css]: - [ ] The card components should be distinguished horizontally as well using either horizontal borders or some diagonal color pattern like in the original design which would improve the UX. :pray: 

image
image

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @aliabbani , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
- [ ] I cannot find a video link either in your PR summary or in the readme file. Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:
- a description of the project. 
- a demo of the project features.
- you should also highlight some interesting piece of code or something you built that you are very proud of.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (aliabbani) [reply README.md]: Hi @henatan99 thank you for your comment, I moved it as you can see in the image below

comment2

File: src/stylesheet/index.css (hunk 1 above)
Comment (aliabbani) [reply src/stylesheet/index.css]: Hi @henatan99 thank you for your comment, I solved it as you can see in the image below

first-comment

File: src/stylesheet/index.css (hunk 4 above)
Comment (aliabbani) [reply src/stylesheet/index.css]: Hi @henatan99 thank you for your comment, I fixed the UI as you can see in the image below

comment4
Capture

File: src/components/Home.js (hunk 3 above)
Comment (aliabbani) [reply src/components/Home.js]: Hi @henatan99 thank you for your comment, I created HomeHeader and HomeCountries components as you can see in the image below

Capture333333

File: src/stylesheet/index.css (hunk 1 above)
Comment (henatan99) [reply src/stylesheet/index.css]: This issue is well addressed. Nice work! :clap: :+1:

File: README.md (hunk 2 above)
Comment (henatan99) [reply README.md]: This issue is well addressed. Good job! :pray:

File: src/stylesheet/index.css (hunk 4 above)
Comment (henatan99) [reply src/stylesheet/index.css]: This issue is well addressed. Great job! :clap: :+1:

File: src/components/Home.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+const Home = (props) => {
+  const { totals } = props;
+  const { datas } = props;
+
+  return (
+    <div className="main-container">
+
+      <HomeHeader totals={totals} />
+      <h5 className="home-title">STATS BY COUNTRY</h5>
+      <HomeCountries datas={datas} />
```
Comment (henatan99) [inline src/components/Home.js:16]: - [x] OPTIONAL: I am very glad you make the requested changes and make use of smaller components here. I also suggest that the `h5` element could have been better inserted inside the `HomeCountries` component. It is not required for this project submission but you can take your time later to make this change. :+1:

**Review by henatan99 — APPROVED** [review]
Hi @aliabbani , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**